sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,366 @@
+    static BlockInfo StageBlockList(
+        // TODO: Context and Pipeline should go here
+        /*const*/ std::string& url,
+        /*const*/ StageBlockListOptions& options)
+    {
+        //TODO Manage the memory that is allocated when constructing the request.
+      return StageBlockListParseResponse(
+          azure::core::http::Client::send(StageBlockListConstructRequest(url, options)));
+    }
+
+  private:
+    class block_list_writer : public Azure::Storage::Common::XML::xml_writer
+    {
+    public:
+      block_list_writer() {}
+
+      std::string write(const std::vector<std::pair<BlockType, std::string>>& blockList)
+      {
+        std::ostringstream outstream;
+        initialize(outstream);
+
+        write_start_element(k_XML_TAG_BLOCK_LIST);
+
+        for (auto block = blockList.cbegin(); block != blockList.cend(); ++block)
+        {
+          std::string tag;
+          switch (block->first)
+          {
+            case BlockType::Committed:
+              tag = k_XML_TAG_COMMITTED;
+              break;
+
+            case BlockType::Uncommitted:
+              tag = k_XML_TAG_UNCOMMITTED;
+              break;
+
+            case BlockType::Latest:
+              tag = k_XML_TAG_LATEST;
+              break;
+          }
+
+          write_element(tag, block->second);
+        }
+
+        finalize();
+        return outstream.str();
+      }
+    };
+  };
+
+private:
+  static void AddMetadata(
+      /*const*/ std::map<std::string, std::string>& metadata,
+      azure::core::http::Request& request)
+  {
+    for (auto pair : metadata)
+    {
+      request.addHeader(k_HEADER_METADATA_PREFIX + pair.first, pair.second);
+    }
+  }
+
+  static void ApplyBasicHeaders(
+      /*const*/ RequestOptions& options,
+      azure::core::http::Request& request)
+  {
+    if (options.Date.empty())
+    {
+      options.Date = GetDateString();
+    }
+    request.addHeader(k_HEADER_X_MS_VERSION, options.Version);
+    request.addHeader(k_HEADER_CLIENT_REQUEST_ID, options.ClientRequestID);
+    request.addHeader(k_HEADER_DATE, options.Date);
+  }
+
+  static void ParseBasicResponseHeaders(
+      const std::map<std::string, std::string>& headers,
+      ResponseInfo& info)
+  {
+    info.RequestId = GetHeaderValue(headers, k_HEADER_X_MS_REQUEST_ID);
+    info.ClientRequestID = GetHeaderValue(headers, k_HEADER_X_MS_CLIENT_REQUEST_ID);
+    info.Version = GetHeaderValue(headers, k_HEADER_X_MS_VERSION);
+    info.Date = GetHeaderValue(headers, k_HEADER_DATE);
+  }
+
+  static std::string GetHeaderValue(
+      const std::map<std::string, std::string>& headers,
+      const std::string& key)
+  {
+    if (headers.find(key) != headers.end())
+    {
+      return headers.at(key);
+    }
+    return std::string();
+  }
+
+  static std::string GetDateString()
+  {
+    char buf[30];
+    std::time_t t = std::time(nullptr);
+    std::tm* pm;
+    pm = std::gmtime(&t);
+    size_t s = std::strftime(
```

Comment:
This is locale sensitive so you will run into the same bug I spent a bunch of time fixing in cpprestsdk for you folks. I recommend borrowing an implementation like this one which avoids the locale impact: https://github.com/microsoft/cpprestsdk/blob/master/Release/src/utilities/asyncrt_utils.cpp#L745